src/H5Ofill.c

```diff
@@ -218,6 +219,11 @@ H5O_fill_new_decode(H5F_t H5_ATTR_UNUSED *f, hid_t H5_ATTR_UNUSED dxpl_id, H5O_t
             INT32DECODE(p, fill->size);
             if (fill->size > 0) {
                 H5_CHECK_OVERFLOW(fill->size, ssize_t, size_t);
+
+                /* Ensure that fill size doesn't exceed buffer size, due to possible data corruption */
+                if (p + fill->size - 1 > p_end)
+                    HGOTO_ERROR(H5E_RESOURCE, H5E_OVERFLOW, NULL, "fill size exceeds buffer size")
```

Comment:
These errors should be H5E_OHDR, not H5E_RESOURCE.

Reply:
I fixed, pushed, but haven't seen it...

Reply:
Thanks.  Fixed.